src/jit/core_templates.expr

```diff
@@ -640,11 +638,9 @@
       (store $0 $obj ptr_sz))))
 
 (template: isconcrete
-  (if (all
-        (nz $1)
-        (^is_conc_obj $1))
-    (const 1 int_sz)
-    (const 0 int_sz)))
+  (and
+    (flagval (nz $1))
+    (flagval (^is_conc_obj $1))))
```

Comment:
This one relied on the 'short-circuit' behaviour of `all` - after the first `nz $1`, the `^is_conc_obj` (which derferences the pointer) isn't executed. The bitwise-version skips that, so this one isn't safe.